scripts/modules/sdk_operations.js

```diff
@@ -0,0 +1,147 @@
+'use strict';
+/*
+ * Operations for processing nuspecs and csproj on SDK projects. 
+ */
+
+const xmlhandling = require('./xml_handling')
+const chalk = require('chalk')
+const promisify = require('util').promisify
+const execFile = promisify(require('child_process').execFile);
+
+/**
+ * Processes project, shows information about it, and bumps version if it is required
+ * @param {Document} doc not used
+ * @param {string} projConfig project configuration
+ * @param {string} projPath path to .csproj 
+ * @param {boolean} dryrun true=no changes to nuspec,false=changes will be written
+ */
+async function processSDKProject(doc, projConfig, projPath, dryrun) {
+    const dir = `${__dirname}/../../`;
+    try {
+        const doc = await xmlhandling.openXML(dir, projConfig["nuspec"])
+        const metadata = doc.documentElement.getElementsByTagName("metadata").item(0)
+        const versionElement = metadata.getElementsByTagName('version').item(0)
+        const version = versionElement.textContent
+        let addendum = ""
+        if (typeof projConfig["bumpVersion"] != 'undefined') {
+            const splitVersion = version.split('.')
+            const suffixSplit = splitVersion[2].split('-')
+            if (suffixSplit.length == 2) {
+                splitVersion[2] = suffixSplit[0]
+                splitVersion[3] = suffixSplit[1]
+            }
+            switch (projConfig["bumpVersion"]) {
+                case 'major':
+                    splitVersion[0]++
+                    splitVersion[1] = 0
+                    splitVersion[2] = 0
+                    break;
+                case 'minor':
+                    splitVersion[1]++
+                    splitVersion[2] = 0
+                    break;
+                case 'patch':
+                    splitVersion[2]++
```

Comment:
how does this work if the version has labels like `1.0.0-alpha`? There is a semver package in npm, might be better using that instead?